src/mono/msbuild/android/build/AndroidBuild.targets

```diff
@@ -40,36 +55,41 @@
         Include="$(RuntimeComponents)" />
 
       <_RuntimeLibraries 
-        Include="$(AndroidAppDir)\*-stub-static.a" />
+        Include="$(AndroidBuildDir)\*-stub-static.a" />
       <_RuntimeLibraries 
-        Include="$(AndroidAppDir)\*.a"
-        Exclude="$(AndroidAppDir)\*-static.a" />
+        Include="$(AndroidBuildDir)\*.a"
+        Exclude="$(AndroidBuildDir)\*-static.a" />
 
-      <_RuntimeLibraries Remove="$(AndroidAppDir)\libmono-component-%(_UsedComponents.Identity)-stub-static.a" />
-      <_RuntimeLibraries Include="$(AndroidAppDir)\libmono-component-%(_UsedComponents.Identity)-static.a" />
+      <_RuntimeLibraries Remove="$(AndroidBuildDir)\libmono-component-%(_UsedComponents.Identity)-stub-static.a" />
+      <_RuntimeLibraries Include="$(AndroidBuildDir)\libmono-component-%(_UsedComponents.Identity)-static.a" />
     </ItemGroup>
   </Target>
 
-  <Target Name="_BeforeAndroidBuildApp">
+  <Target Name="_BeforeAndroidBuild">
     <PropertyGroup>
-      <_AndroidRuntimeConfigFilePath Condition="'$(_AndroidRuntimeConfigFilePath)' == ''">$([MSBuild]::NormalizePath($(AndroidAppDir), '$(AssemblyName).runtimeconfig.json'))</_AndroidRuntimeConfigFilePath>
-      <_ParsedRuntimeConfigFilePath Condition="'$(_ParsedRuntimeConfigFilePath)' == ''">$([MSBuild]::NormalizePath($(AndroidAppDir), 'runtimeconfig.bin'))</_ParsedRuntimeConfigFilePath>
+      <_AndroidRuntimeConfigFilePath Condition="'$(_AndroidRuntimeConfigFilePath)' == ''">$([MSBuild]::NormalizePath($(AndroidBuildDir), '$(AssemblyName).runtimeconfig.json'))</_AndroidRuntimeConfigFilePath>
+      <_ParsedRuntimeConfigFilePath Condition="'$(_ParsedRuntimeConfigFilePath)' == ''">$([MSBuild]::NormalizePath($(AndroidBuildDir), 'runtimeconfig.bin'))</_ParsedRuntimeConfigFilePath>
     </PropertyGroup>
 
-    <RemoveDir Directories="$(AndroidAppBundleDir)" />
+    <RemoveDir Directories="$(AndroidBundleDir)" />
   </Target>
 
   <Target Name="_AndroidResolveReferences">
+    <ItemGroup Condition="'$(EnableDefaultAssembliesToBundle)' == 'true'">
+      <AndroidAssembliesToBundle Remove="@(AndroidAssembliesToBundle)" />
+      <AndroidAssembliesToBundle Include="$(PublishDir)\**\*.dll" />
+    </ItemGroup>
+
     <ItemGroup>
-      <AppAssembliesInternal Remove="@(AppAssembliesInternal)" />
-      <AppAssembliesInternal Include="@(AndroidAssembliesToBundle)">
+      <AssembliesToBundleInternal Remove="@(AssembliesToBundleInternal)" />
```

Comment:
In most properties we prefix with _ to make them internal, should we apply the same schema for this property as well?

Reply:
Fixed